tsl/test/expected/continuous_aggs_union_view-10.out

```diff
@@ -0,0 +1,393 @@
+-- This file and its contents are licensed under the Apache License 2.0.
+-- Please see the included NOTICE for copyright information and
+-- LICENSE-APACHE for a copy of the license.
+-- disable background workers to make results reproducible
+\c :TEST_DBNAME :ROLE_SUPERUSER
+SELECT _timescaledb_internal.stop_background_workers();
+ stop_background_workers 
+-------------------------
+ t
+(1 row)
+
+\c :TEST_DBNAME :ROLE_DEFAULT_PERM_USER
+-- look at postgres version to decide whether we run with analyze or without
+SELECT
+  CASE WHEN current_setting('server_version_num')::int >= 100000
+    THEN 'EXPLAIN (analyze, costs off, timing off, summary off)'
+    ELSE 'EXPLAIN (costs off)'
+  END AS "PREFIX"
+\gset
+CREATE TABLE metrics(f1 int, f2 int, time timestamptz NOT NULL, device_id int, value float);
+SELECT create_hypertable('metrics','time');
+  create_hypertable   
+----------------------
+ (1,public,metrics,t)
+(1 row)
+
+ALTER TABLE metrics DROP COLUMN f1;
+INSERT INTO metrics(time, device_id, value) SELECT '2000-01-01'::timestamptz, device_id, 0.5 FROM generate_series(1,3) g(device_id);
+--
+-- test switching continuous agg view between different modes
+--
+-- check default view for new continuous aggregate
+CREATE VIEW metrics_summary
+  WITH (timescaledb.continuous)
+AS
+  SELECT time_bucket('1d',time), avg(value) FROM metrics GROUP BY 1;
+ALTER TABLE metrics DROP COLUMN f2;
```

Comment:
What would happen if you drop a column which is 'SELECT'ed (used) in the view? Would the view updated or would it fail (I guess the latter)?

I suggest making column f2 part of the view before trying to drop it to test this case. For instance, make the view query become: `SELECT time_bucket('1d',time), f2, avg(value) FROM metrics GROUP BY 1,2;`

Reply:
you cant drop a column from a table that is referenced by any view unless you use CASCADE which would drop the view too